pallets/tx-pause/src/lib.rs

```diff
@@ -135,21 +142,200 @@ pub mod pallet {
             pallet_name: Vec<u8>,
             function_name: Vec<u8>,
         ) -> DispatchResult {
-            T::UpdateOrigin::ensure_origin(origin)?;
-            if PausedTransactions::<T>::take((&pallet_name, &function_name)).is_some() {
-                Self::deposit_event(Event::TransactionUnpaused(pallet_name, function_name));
-            };
+            T::UnpauseOrigin::ensure_origin(origin)?;
+
+            Self::unpause_one(&pallet_name, &function_name)?;
+
             Ok(())
         }
+
+        /// Pause extrinsics by passing the extrinsic and corresponding pallet names.
+        /// Use names as they are written in the source code of the pallet.
+        #[pallet::call_index(2)]
+        #[pallet::weight({
+            let len = pallet_and_funcs.iter().flat_map(|item| {item.clone().1}).count();
+            T::WeightInfo::pause_transaction().saturating_mul(len as Weight)
+        })]
+        #[transactional]
+        pub fn pause_transactions(
+            origin: OriginFor<T>,
+            pallet_and_funcs: Vec<(Vec<u8>, Vec<Vec<u8>>)>,
+        ) -> DispatchResult {
+            T::PauseOrigin::ensure_origin(origin)?;
+
+            for (pallet_name, function_name) in pallet_and_funcs {
+                Self::ensure_can_pause(&pallet_name)?;
+
+                for call_name in function_name {
+                    Self::pause_one(&pallet_name, &call_name, true)?;
+                }
+            }
+
+            Ok(())
+        }
+
+        /// Unpause extrinsics by passing the extrinsic and corresponding pallet names.
+        /// Use names as they are written in the source code of the pallet.
+        #[pallet::call_index(3)]
+        #[pallet::weight({
+            let len = pallet_and_funcs.iter().flat_map(|item| {item.clone().1}).count();
+            T::WeightInfo::unpause_transaction().saturating_mul(len as Weight)
+        })]
+        #[transactional]
+        pub fn unpause_transactions(
+            origin: OriginFor<T>,
+            pallet_and_funcs: Vec<(Vec<u8>, Vec<Vec<u8>>)>,
+        ) -> DispatchResult {
+            T::UnpauseOrigin::ensure_origin(origin)?;
+
+            for (pallet_name, function_name) in pallet_and_funcs {
+                for call_name in function_name {
+                    Self::unpause_one(&pallet_name, &call_name)?;
+                }
+            }
+
+            Ok(())
+        }
+
+        /// Pause all the calls of the listed pallets in `pallet_names`.
+        /// This logic is in its own extrinsic in order to not have to pause calls 1 by 1.
+        #[pallet::call_index(4)]
+        #[pallet::weight({
+            let len = pallet_names.len();
+            let max = T::MaxCallNames::get();
+            T::WeightInfo::pause_transaction().saturating_mul(len as Weight).saturating_mul(max as Weight)
```

Comment:
I don't really understand what is this annotation used for, when we have an explicit return at the bottom.

Reply:
right, documentation here: https://docs.substrate.io/build/tx-weights-fees/#post-dispatch-weight-correction. But I need manual test to make sure it works. As current we're using root which can't test this scene. I'll try set to other account.